### DISCUSSION TITLE
Is it correct to assume  k_score_ as a result?

### DISCUSSION THREAD

**Discussion by henrique-voni** [rasbt/mlxtend#812] [Q&A]
Hi, i am running some experiments using mlxtend with the `SequentialFeatureSelector` and i have a question. This is my current pipeline using SFS:

1) Split data into train/test
2) Run `SequentialFeatureSelector` over train set and get the best subset of features
3) Transform train & test sets with `SequentialFeatureSelector.transform()` to select the best subset of features
4) Fit a new model (same as used in SFS) with transformed train data
5) Evaluate predicting the transformed test data

This approach is based in the documentation example found [here](http://rasbt.github.io/mlxtend/user_guide/feature_selection/SequentialFeatureSelector/#example-7-using-the-selected-feature-subset-for-making-new-predictions), however i see that SequentialFeatureSelector is able to perform cross-validation during the feature selection process (the `cv` parameter) and already gives the best k_score found during the process. 

My question is: instead of applying the SFS over the train set only, would it be correct to pass the entire dataset and set a value for `cv` and then just assume the `k_score_` as an evaluation metric for my experiment? 

Thanks in advance!

**Comment by rasbt** [comment]
Hi there,

your 5-step approach sounds reasonable. Ideally, the test set should be treated as independent dataset that is not involved in the modeling process, so I wouldn't do the feature selection on the entire dataset.

> however i see that SequentialFeatureSelector is able to perform cross-validation during the feature selection process (the cv parameter) and already gives the best k_score found during the process

That's correct, but it is primarily such that it doesn't overfit so heavily compared to not using cross-validation on the training set.
